src/main/java/com/redhat/quarkus/mandrel/collector/report/model/ImageStatsCollection.java

```diff
@@ -70,17 +73,68 @@ public ImageStats add(ImageStats stat, String tag, Long runnerInfoId) {
         return stat;
     }
 
-    public ImageStats[] getAll() {
-        return em.createNamedQuery("ImageStats.findAll", ImageStats.class).getResultList().toArray(new ImageStats[0]);
+    public ImageStats[] getAll(String newerThan, String olderThan) {
+        if (StringUtil.isNullOrEmpty(olderThan) && StringUtil.isNullOrEmpty(newerThan)) {
+            return em.createNamedQuery("ImageStats.findAll", ImageStats.class)
+                    .getResultList().toArray(new ImageStats[0]);
+        }
+        DateRange dateRange = DateRange.fromString(newerThan, olderThan);
+        return em.createNamedQuery("ImageStats.findAllByDate", ImageStats.class)
+                .setParameter("range_start", dateRange.start)
+                .setParameter("range_end", dateRange.end)
+                .getResultList().toArray(new ImageStats[0]);
     }
 
     public ImageStats[] getAllByTag(String tag) {
-        return em.createNamedQuery("ImageStats.findByTag", ImageStats.class).setParameter("tag_name", tag)
+        return getAllByTag(null, null, tag);
+    }
+
+    public ImageStats[] getAllByTag(String newerThan, String olderThan, String tag) {
+        if (StringUtil.isNullOrEmpty(olderThan) && StringUtil.isNullOrEmpty(newerThan)) {
+            return em.createNamedQuery("ImageStats.findByTag", ImageStats.class)
+                    .setParameter("tag_name", tag)
+                    .getResultList().toArray(new ImageStats[0]);
+        }
+        DateRange dateRange = DateRange.fromString(newerThan, olderThan);
```

Comment:
Same here. You are checking for `olderThan != null && newerThan != null` but `olderThan == null || newerThan == null` might still be `true`.